src/github.com/matrix-org/dendrite/roomserver/storage/events_table.go

```diff
@@ -194,7 +194,9 @@ func (s *eventStatements) bulkSelectStateEventByID(eventIDs []string) ([]types.S
 		// However it should be possible debug this by replaying queries or entries from the input kafka logs.
 		// If this turns out to be impossible and we do need the debug information here, it would be better
 		// to do it as a separate query rather than slowing down/complicating the common case.
-		return nil, fmt.Errorf("storage: state event IDs missing from the database (%d != %d)", i, len(eventIDs))
+		return nil, types.MissingEventError(
```

Comment:
As per the doc on [error handling in dendrite](https://docs.google.com/document/d/1VUkAafJgk-e0S4zjBah6t9Oriw-WUgM-76Vl7iDyeIQ/edit) - please document the public functions which can return a `types.MissingEventError`.